zebra-state/src/service/check/nullifier.rs

```diff
@@ -63,10 +60,14 @@ pub(crate) fn no_duplicates_in_finalized_chain(
 /// (Duplicate finalized nullifiers are rejected during service contextual validation,
 /// see [`no_duplicates_in_finalized_chain`] for details.)
 ///
-/// "A transaction is not valid if it would have added a nullifier
-/// to the nullifier set that already exists in the set"
+/// # Consensus
+///
+/// > A nullifier MUST NOT repeat either within a transaction,
+/// > or across transactions in a valid blockchain.
+/// > Sprout and Sapling and Orchard nullifiers are considered disjoint,
+/// > even if they have the same bit pattern.
 ///
-/// https://zips.z.cash/protocol/protocol.pdf#commitmentsandnullifiers
+/// https://zips.z.cash/protocol/protocol.pdf#nullifierset
 #[tracing::instrument(skip(chain_nullifiers, shielded_data_nullifiers))]
 pub(crate) fn add_to_non_finalized_chain_unique<'block, NullifierT>(
```

Comment:
From the implementation of the function, it's not clear that 
> Sprout and Sapling and Orchard nullifiers are considered disjoint, even if they have the same bit pattern.

I checked how the function is used, and the code that wraps this function uses it so that the nullifier sets are disjoint.

Reply:
Clarified in fd99912b8